src/sage/graphs/generators/random.py

```diff
@@ -1556,22 +1567,45 @@ def RandomPartialKTree(n, k, x, seed=None):
         5
         sage: G.show()  # not tested
     """
+    if n < 0:
+        raise ValueError("n must not be negative")
+
+    if k < 0:
+        raise ValueError("k must not be negative")
+
+    # A graph with treewidth 0 has no edges
+    if k == 0:
+        g = Graph(n - k, name=f"Random Partial 0-tree of order {n - x}")
+        return g
+
+    if n < k + 1:
+        raise ValueError("n must be greater than k")
+
     if seed is not None:
         set_random_seed(seed)
 
-    g = RandomKTree(n, k, seed)
+    # This formula calculates how many edges are in a `k`-tree with `n` nodes
+    edgesInKTree = (k ^ 2 + k) / 2 + (n - k - 1) * k
 
     # Check that x doesn't delete too many edges
-    # This formula calculates how many edges are in `k`-tree with `n` nodes
-    if x > (k ^ 2 + k) / 2 + (n - k - 1) * k:
+    if x > edgesInKTree:
         raise ValueError("x must be less than the number of edges in the `k`-tree with `n` nodes")
 
+    # The graph will have no edges
+    if x == edgesInKTree:
+        g = Graph(0, name=f"Random Partial {k}-tree of order 0")
```

Comment:
Is this the expected behavior ? For me the graph has `n` vertices, not 0. Indeed, when you remove edges you may obtain isolated vertices, but these vertices remain in the graph.

Reply:
My mistake, you are correct